packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -766,13 +769,17 @@ function completeWork(
       break;
     }
     case EventComponent: {
-      const rootContainerInstance = getRootHostContainer();
-      const responder = workInProgress.type.responder;
-      handleEventComponent(responder, rootContainerInstance, workInProgress);
+      if (enableEventAPI) {
+        const rootContainerInstance = getRootHostContainer();
+        const responder = workInProgress.type.responder;
+        handleEventComponent(responder, rootContainerInstance, workInProgress);
+      }
       break;
     }
     case EventTarget: {
-      markUpdate(workInProgress);
+      if (enableEventAPI) {
+        markUpdate(workInProgress);
```

Comment:
This is fine as a start but longer term it would be good if we could avoid doing this in the commit phase for initial renders.

For host components we add events and stuff during render phase during initial mount (`current === null`). We only do stuff in the commit phase for updates.

Reply:
In this case, the event target needs to mutate the direct DOM elements of the event target. This is only known till commit phase as the children might not have been created/attached in complete phase. Well at least that's the issue I ran into when testing this, maybe I'm missing something though.